### ISSUE TITLE
Use of PyGgis class:QgsCoordinateTransform, method:transformCoords() 

### ISSUE THREAD

**Issue by MrBeee** [qgis/QGIS#55028]
###  ```QgsCoordinateTransform.transformCoords()```  does not work as intended/documented

The documentation mentions that ```transformCoords()``` uses an int to set the number of coordinates to be transformed, together with three arrays, containing x, y, & z data, followed by a direction flag.

 ```transformCoords()```  is supposed to return a tuple of some sorts, though the square brackets in the documentation 'hint' at a list

The problem is that Python does not know 'arrays' as a native data type. 

- So I tried using 1D numpy arrays. This does not work
- Next I tried using lists. This does not work either

Therefore I checked the source code on GitHub, to find out that the native c++ ```transformCoords()```  routine:

- uses c++ arrays as input. There is no comparable data type in Python, unless one starts to 'hack' numpy arrays to access the underlying buffers in the numpy array
- is of type ```void```.  Hence it does not return any transformed data at all. From this it follows that the routine does in-place transforms. But the documentation says otherwise.

### Bottom line 

It would be great if someone can provide an example how to work with ```transformCoords()```  in the Python environment. 

Reason: transforming many coordinates at once leads to (much) less overhead in the transform. But I haven't been able to find such an example, and looking at the source code, it seems the Python bindings don't work as expected/documented.

Cheers,

Bart

**Comment by MrBeee** [comment]
Indeed Nicolas; it is more than "just a documentation issue". 
Has this issue/bug been allocated to someone ?

A pure **documentation issue** is with ```transformInPlace```, that mentions:
```transformInPlace(self, direction: [Qgis.TransformDirection] = Qgis.TransformDirection.Forward) → Tuple[float, float, float]```

Please note: 
- The text ```→ Tuple[float, float, float]``` implies that the function returns with a value, which isn't the case....
- The parameters x, y, z are missing from the function description

Hope this can be fixed.

Cheers,

Bart

**Comment by agiudiceandrea** [comment]
Has anyone found a way to properly use QgsCoordinateTransform transformCoords and transformInPlace in Python?

**Comment by ViperMiniQ** [comment]
I am not sure if `transformCoords` can be fixed with simple annotations:

https://python-sip.readthedocs.io/en/stable/annotations.html#argument-annotation-Array
https://python-sip.readthedocs.io/en/stable/annotations.html#argument-annotation-ArraySize

If I am not mistaken, we would need to define this 3 times, but we can do it only once(?)
